### ISSUE TITLE
SfM camera does not work as expected

### ISSUE THREAD

**Issue by xuxy09** [facebookresearch/pytorch3d#171]
I am trying to render a textured human with the SfM perspective camera model of Pytorch3d, however I cannot get any meaningful output. The code to reproduce my result is here: https://github.com/xuxy09/pytorch3d_test/blob/master/pytorch3d_render.py

The obj file is produced by SMPL model and works well with blender and Opendr.

Please let me know how to fix this problem. Thanks!

**Comment by gkioxari** [comment]
Could you provide the `.obj` file as well so that we can debug this?

**Comment by xuxy09** [comment]
The obj is also included in the repo: https://github.com/xuxy09/pytorch3d_test

Thanks.

**Comment by xuxy09** [comment]
Hi,

The obj is also included in the repo:
https://github.com/xuxy09/pytorch3d_test

Thanks.


Best

Georgia Gkioxari <notifications@github.com> 于2020年4月28日周二 上午11:01写道：

**Comment by gkioxari** [comment]
I think the issue lies with the fact that you specify the camera in screen coordinates, while it needs to be in ndc coordinates.

**Comment by xuxy09** [comment]
Is there a function in Pytorch3d that we can use to transform the
coordinates?

Georgia Gkioxari <notifications@github.com> 于2020年4月28日周二 下午12:04写道：

**Comment by gkioxari** [comment]
Hi @xuxy09 

We will add support for conversions between world to ndc to screen. These transforms are not hard to figure out. I provide the solution for your case though to help you out. Replace the `cameras` with the code below: 

```
T = torch.tensor([-0.0578, 0.1563, 39.0994]).view(1, -1).to(device)
R = torch.eye(3, device=device).view(1, 3, 3)
# flip y axis so that human appears upright. This is because PyTorch3D world's system
# assumes (+X left, +Y up, +Z from you to monitor). Alternatively, you can skip the following 
# lines and you can convert your mesh to our world coordinate system after loading. 
R[0, 1, 1] *= -1.0
T[1] *= -1.0
# focal_length is set to 44.6 (which equals your focal_length / half_image_size = 5000 / 112)
cameras = SfMPerspectiveCameras(focal_length=44.6, device=device, R=R, T=T)
```

The output I get is shown below
![rendered_output](https://user-images.githubusercontent.com/4369065/80524412-68f6af00-8944-11ea-9c59-7aebd81ad173.png)

**Comment by xuxy09** [comment]
I still cannot get a reasonable output with your code. Did you accidentally
change anything else? By the way, there seems a small bug in your code: T[1]
*= -1.0 should be T[0] *= -1.0?

Georgia Gkioxari <notifications@github.com> 于2020年4月28日周二 下午2:36写道：

**Comment by gkioxari** [comment]
No I didn't change anything else. The rest of the code is exactly what you have. 

The right transformation is `T[0, 1] *= -1`, because you only want to change the y-axis. `T[0] *=-1` flips ALL axes. You don't want that.

**Comment by xuxy09** [comment]
It finally worked. Thanks a lot!
